mne_bids_pipeline/tests/test_run.py

```diff
@@ -278,3 +282,131 @@ def test_missing_sessions(
+        new_root = walk_root
+        if "sub-01" in walk_root.parts or "sub-02" in walk_root.parts:
+            new_root = Path(
+                *[freesurfer_subject_mapping.get(p, p) for p in new_root.parts]
+            )
+        offset = new_root.relative_to(src_dir)
+        # the actual subject dirs need their names changed
+        for _dir in dirs:
+            _dir = freesurfer_subject_mapping.get(_dir, _dir)
+            (dst_dir / offset / _dir).mkdir()
+        # for filenames that contain the subject identifier (BEM files, morph maps),
+        # we need to change the filename too, not just parent folder name
+        for _file in files:
+            dst_file = _file
+            for subj in freesurfer_subject_mapping:
+                if subj in dst_file:
+                    dst_file = dst_file.replace(subj, freesurfer_subject_mapping[subj])
+                    break
+            shutil.copyfile(src=walk_root / _file, dst=dst_dir / offset / dst_file)
+    # print_dir_tree(new_bids_path.root)  # for debugging
+    # update config so that `subjects_dir` and `deriv_root` also point to the tempdir
+    extra_config = f"""
+from pathlib import Path
+subjects_dir = "{new_bids_path.root / "derivatives" / "freesurfer" / "subjects"}"
+deriv_root = Path("{new_bids_path.root}") / "derivatives" / "mne-bids-pipeline" / "MNE-funloc-data"
+"""  # noqa E501
+    extra_path = tmp_path / "extra_config.py"
+    extra_path.write_text(extra_config)
+    monkeypatch.setenv("_MNE_BIDS_STUDY_TESTING_EXTRA_CONFIG", str(extra_path))
+    # Run the tests.
+    steps = test_options.get("steps", ())
+    command = ["mne_bids_pipeline", str(config_path), f"--steps={','.join(steps)}"]
+    # hack in the new bids_root
+    command.append(f"--root-dir={new_bids_path.root}")
+    if "--pdb" in sys.argv:
+        command.append("--n_jobs=1")
+    monkeypatch.setenv("_MNE_BIDS_STUDY_TESTING", "true")
+    monkeypatch.setattr(sys, "argv", command)
+    with capsys.disabled():
+        print()
+        main()
+    # check some things that are indicative of different MRIs being used in each session
+    results = list()
+    for sess in ("a", "b"):
+        fname = (
+            new_bids_path.root
+            / "derivatives"
+            / "mne-bids-pipeline"
+            / "MNE-funloc-data"
+            / "sub-01"
+            / f"ses-{sess}"
+            / "meg"
+            / f"sub-01_ses-{sess}_task-funloc_report.h5"
+        )
+        report = read_hdf5(fname, title="mnepython")
+        coregs = next(
+            filter(lambda x: x["dom_id"] == "Sensor_alignment", report["_content"])
+        )
+        pattern = re.compile(
+            r"Average distance from (?P<npts>\d+) digitized points to head: "
+            r"(?P<dist>\d+(?:\.\d+)?) mm"
+        )
+        result = pattern.search(coregs["html"])
+        assert result is not None
+        assert float(result.group("dist")) < 3
```

Comment:
```suggestion
        assert float(result.group("dist")) < 3  # fit between pts and outer_skin < 3 mm
```
add an explainer comment